opout

```diff
@@ -2,9 +2,9 @@
 # opout: "open output": A general handler for opening a file's intended output.
 # I find this useful especially running from vim.
 
-basename="$(echo "$1" | sed 's/\.[^\/.]*$//')"
+basename="${1%.*}"
```

Comment:
I think this is actually a misleading variable name. That, AFAIK, is not the basename, but I guess "filename with its extension stripped" is a bit much for a variable. :P

But I think what Brodie is shooting for here, is to omit the extension, but not therefore result in an empty filename, because of filenames which are hidden (prefixed with `.`) and have _no_ extension at all. Your change here would result in an empty string for a hidden file with an extension, basically. However, neither the original nor your code appears to be fool-proof, but I would, just in-case, opt for:

```suggestion
# Avoid extension-less hidden filenames from being read as null. 
basename_nodot=${1#.}

basename=${basename_nodot%.*}
```